src/providers/ad/ad_gpo.c

```diff
@@ -2179,15 +2178,6 @@ ad_gpo_target_dn_retrieval_done(struct tevent_req *subreq)
                              host_attrs, &msg);
     if (ret == ENOENT) {
         /* The computer is not in cache so query LDAP server */
-        /* Convert the domain name into domain DN */
-        ret = domain_to_basedn(state, state->host_domain->name, &domain_dn);
-        if (ret != EOK) {
-            DEBUG(SSSDBG_OP_FAILURE,
-                  "Cannot convert domain name [%s] to base DN [%d]: %s\n",
-                   state->host_domain->name, ret, sss_strerror(ret));
-            goto done;
-        }
-
         subreq = sdap_get_generic_send(state, state->ev, state->opts,
                                        sdap_id_op_handle(state->sdap_op),
                                        state->target_dn, LDAP_SCOPE_BASE,
```

Comment:
If state->target_dn should not be domain_dn in this line?

Reply:
No, this is not a generic search. We are retrieving a specific object (scope=base returns only the target_dn object).

Reply:
No, this is not a generic search. We are retrieving a specific object (scope=base returns only the target_dn object).